operators/pkg/instance-controller/controller.go

```diff
@@ -140,9 +170,23 @@ func (r *InstanceReconciler) generateEnvironments(template *crownlabsv1alpha2.Te
 	return ctrl.Result{}, nil
 }
 
+func ignoreUpdatePredicate() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if Annot := e.MetaOld.GetAnnotations(); Annot != nil {
+				if value, ok := Annot["NeedToReconcile"]; ok {
+					return value == "True"
+				}
+			}
+			return true
+		},
+	}
+}
```

Comment:
We have tried without this function and it works, it was useful with previous implementation so now we will delete it.
Since we will delete this function the needToReconcile filter is useless